src/lib.rs

```diff
@@ -141,82 +141,92 @@ pub fn consumer(
 ) {
     let mut gcov_type = GcovType::Unknown;
 
-    while let Some(work_item) = queue.pop() {
-        let new_results = match work_item.format {
-            ItemFormat::GCNO => {
-                match work_item.item {
-                    ItemType::Path(gcno_path) => {
-                        // GCC
-                        run_gcov(&gcno_path, branch_enabled, working_dir);
-                        let gcov_path =
-                            gcno_path.file_name().unwrap().to_str().unwrap().to_string() + ".gcov";
-                        let gcov_path = working_dir.join(gcov_path);
-                        if gcov_type == GcovType::Unknown {
-                            gcov_type = if gcov_path.exists() {
-                                GcovType::SingleFile
-                            } else {
-                                GcovType::MultipleFiles
-                            };
-                        }
-
-                        if gcov_type == GcovType::SingleFile {
-                            let new_results = try_parse!(parse_gcov(&gcov_path), work_item.name);
-                            fs::remove_file(gcov_path).unwrap();
-                            new_results
-                        } else {
-                            let mut new_results: Vec<(String, CovResult)> = Vec::new();
-
-                            for entry in WalkDir::new(&working_dir).min_depth(1) {
-                                let gcov_path = entry.unwrap();
-                                let gcov_path = gcov_path.path();
-
-                                new_results.append(&mut try_parse!(
-                                    parse_gcov(&gcov_path),
-                                    work_item.name
-                                ));
-
-                                fs::remove_file(gcov_path).unwrap();
+    loop {
+        // If the queue is empty then it means that the producer
+        // doesn't push anything for the moment so it loops..
```

Comment:
This is not great, as we are spin-waiting here uselessly consuming resources since the pop is not blocking.